### DISCUSSION TITLE
ResourceLoader consistenly throws an exception of type 'System.TypeInitializationException'

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#16328] [Q&A]
Context: Uno4; VS2o22Community; Windows11

I had adapted code in the [Switching to string resources](https://platform.uno/docs/articles/guides/silverlight-migration/05-string-resources.html) (specifically, in the [Reading string resources in code](https://platform.uno/docs/articles/guides/silverlight-migration/05-string-resources.html#reading-string-resources-in-code) section) to store some JavaScript locally rather than depend on the internet to be always present. 

The code I came up with for reading the custom RESX resource in String\en\JavaScriptLibraries.resx
```cs
using Windows.ApplicationModel.Resources;

namespace PelotonIDE.Presentation
{
    public static class JavaScriptLibrariesHelper
    {
        static ResourceLoader _resourceLoader = ResourceLoader.GetForCurrentView("JavaScriptLibraries");
        public static string GetJavaScriptLibrariesResource(string name)
        {
            return _resourceLoader.GetString(name);
        }
    }
}
```
In discord earlier problems with this were said to be able to be worked around using [#11899](https://github.com/unoplatform/uno/issues/11899), however I didn't find it particularly helpful.

Using the above code consistently fails with a COM error of some sort, namely
![image](https://github.com/unoplatform/uno/assets/209621/fcefb172-b26d-4c3d-b73c-37f29f9c1224)

**Comment by jeromelaban** [comment]
Thanks, my answer is similar as on Discord, that we're working on getting https://github.com/unoplatform/uno/pull/10948 done, but in the meantime, you can use workaround here: https://github.com/unoplatform/uno/issues/11899.

We've also just merged https://github.com/unoplatform/uno/pull/16287, but that won't help you as you are having the exception on Windows.

If you've applied the workaround suggested in the issue, please post the new exception.

**Reply by axtens** [reply]
I created an extra helper function, thinking perhaps that the issue was the difference between `.resx` and `.resw`. I copied the symbols in the `.resx` into the default `.resw`. Sadly I still got the same result. As I stepped through the call to `GetResource` (which I am assuming addresses the `.resw` resource) the highlight moved to the `#else` code even though the `#if` code is syntax-coloured, viz
![image](https://github.com/unoplatform/uno/assets/209621/322dafa8-9e82-4fff-95f6-e9ecdd61b5c3)

When calling what I assume to be a routine that tries to load the `.resx` file, I am struggling to know what value to give to `.ResourceLoader()` as nothing works and I get a path not found error, viz
![image](https://github.com/unoplatform/uno/assets/209621/2d46d350-d5f4-4d83-a0f8-c2d0f7f799ae)

The full source for the project is at https://github.com/axtens/PelotonIDE and the file where the `JavaScriptLibrariesHelper` is implemented is `PelotonIDE\PelotonIDE\Presentation\JavaScriptLibrariesHelper.cs`

**Reply by axtens** [reply]
Further oddities:
* if I try to access `new Microsoft.Windows.ApplicationModel.Resources.ResourceLoader("PelotonIDE/Strings")` I get an error message of `"The system cannot find the file specified.\r\n\r\nThe system cannot find the file specified.\r\n"` 
* if I try to access `new Microsoft.Windows.ApplicationModel.Resources.ResourceLoader("PelotonIDE/Strings/en/JavaScriptLibraries")` I get an error message of `"Attempted to access a path that is not on the disk."`

Why the difference?

**Reply by axtens** [reply]
Playing with Uri doesn't help:
`new Uri("ms-appx:///PelotonIDE/Strings/en/JavaScriptLibraries.resx").ToString()` gives me an error and and empty string for the error.Message.
